provisioning/hare_setup.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import argparse
+import asyncio
+import json
+import logging
+import shutil
+import subprocess
+import os
+import yaml
+import socket
+
+from cortx.utils.product_features import unsupported_features
+
+def execute_command_and_get_output(cmd):
+        process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        shell=True)
+        stdout, stderr = process.communicate()
```

Comment:
stdout, _ = process.communicate() ?

Reply:
Exit code is not checked. So if the invoked command returns non-zero, the problem can be swallowed by mistake.

Please have a look how it works in pcswrap: https://github.com/knekrasov/cortx-ha/blob/main/ha/pcswrap/pcswrap/internal/connector.py#L72 
Probably you can follow the same idea.

Reply:
Done